oci/spec_opts.go

```diff
@@ -1653,3 +1653,27 @@ func WithCDIDevices(devices ...string) SpecOpts {
 		return nil
 	}
 }
+
+func WithWasmLayers(descriptors []v1.Descriptor) SpecOpts {
```

Comment:
Another option I considered was to do `WithOciArtifacts`

Reply:
Using `[]v1.Descriptor` directly seems a bit blatant (I'm not sure if that's the right word, so forgive me if it's not), and I'm also thinking about whether there are better function names and arguments

Reply:
Using []v1.Descriptor directly seems a bit blatant (I'm not sure if that's the right word, so forgive me if it's not), and I'm also thinking about whether there are better function names and arguments

Reply:
I am open to suggestions

Reply:
The descriptors seems like exactly what we need here since what we want is for the shim to pull these out of the content store. The function needs a list of mediatypes+digests.

re: function name: `WithAnnotateWasmObjects`?

Reply:
maybe I could pass the whole manifest?  Is that what you mean @Iceber?

that would also allow for usage of the `artifacttype` field on which maybe something that gets set in the future when oci 1.1 is released.